### PR TITLE
Add validation

### PR DESCRIPTION
**Pull request by hub-cap**

### CODE REVIEW

**Comment by TimSimpsonR** [comment]
+1

File: reddwarf/extensions/mysql.py [hunk 1]
```diff
@@ -43,23 +44,35 @@ def get_updated(self):
...
+                              wsgi.ReddwarfXMLDictSerializer()})
         resource = extensions.ResourceExtension(
             'databases',
             service.SchemaController(),
             parent={'member_name': 'instance',
-                     'collection_name': '{tenant_id}/instances'})
+                     'collection_name': '{tenant_id}/instances'},
+            deserializer=wsgi.RequestDeserializer(),
+            serializer=serializer)
+        LOG.info(" yea booooy")
```
Comment (rnirmal) [inline reddwarf/extensions/mysql.py]: yea booooy

File: reddwarf/common/extensions.py [hunk 2]
```diff
@@ -16,62 +16,73 @@
...
+class ReddwarfExtensionMiddleware(extensions.ExtensionMiddleware):
+
+    def __init__(self, application, config, ext_mgr=None):
+        ext_mgr = ext_mgr or ExtensionManager(
+                                 config['api_extensions_path'])
+        mapper = routes.Mapper()
+
+        # extended resources
+        for resource_ext in ext_mgr.get_resources():
+            LOG.debug(_('Extended resource: %s'), resource_ext.collection)
```
Comment (ed-) [inline reddwarf/common/extensions.py:41]: `_(i18n)`

it begins

**Comment by cp16net** [comment]
+1

File: reddwarf/extensions/mysql/service.py [hunk 3]
```diff
@@ -78,10 +118,11 @@ def create(self, req, body, tenant_id, instance_id):
         context = rd_context.ReddwarfContext(
                           auth_tok=req.headers["X-Auth-Token"],
                           tenant=tenant_id)
+        self.validate(body)
```
Comment (rnirmal) [inline reddwarf/extensions/mysql/service.py:118]: can we switch to using the annotations aka decorators

File: reddwarf/extensions/mysql/service.py [hunk 4]
```diff
@@ -117,10 +166,11 @@ def create(self, req, body, tenant_id, instance_id):
...
         schemas = body['databases']
         model_schemas = models.populate_databases(schemas)
         models.Schema.create(context, instance_id, model_schemas)
```
Comment (rnirmal) [inline reddwarf/extensions/mysql/service.py:172]: same here

**Comment by rnirmal** [comment]
+1

**Pull request merged**